actors/builtin/miner/miner_actor.go

```diff
@@ -142,70 +140,40 @@ func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams
 }
 
 //////////////////
-// SurprisePoSt //
+// FallbackPoSt //
 //////////////////
 
-// Called by Actor to notify StorageMiner of SurprisePoSt Challenge.
-func (a Actor) OnSurprisePoStChallenge(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
-	rt.ValidateImmediateCallerIs(builtin.StoragePowerActorAddr)
-
+// Invoked by miner's worker address to submit their fallback post
+func (a Actor) SubmitFallbackPoSt(rt Runtime, params *abi.OnChainPoStVerifyInfo) *adt.EmptyValue {
 	var st State
-	challenged := rt.State().Transaction(&st, func() interface{} {
-		// If already challenged, do not challenge again.
-		// Failed PoSt will automatically reset the state to not-challenged.
-		if st.PoStState.isChallenged() {
-			return false
-		}
+	rt.State().Transaction(&st, func() interface{} {
+		rt.ValidateImmediateCallerIs(st.Info.Worker)
 
-		st.PoStState = PoStState{
-			LastSuccessfulPoSt:     st.PoStState.LastSuccessfulPoSt,
-			SurpriseChallengeEpoch: rt.CurrEpoch(),
-			NumConsecutiveFailures: st.PoStState.NumConsecutiveFailures,
+		if rt.CurrEpoch() > st.PoStState.ProvingPeriodStart+power.FallbackPostChallengeDuration {
+			rt.Abort(exitcode.ErrIllegalState, "PoSt submission too late")
 		}
-		return true
-	}).(bool)
 
-	if challenged {
-		// Request deferred Cron check for SurprisePoSt challenge expiry.
-		cronPayload := CronEventPayload{
-			EventType: CronEventType_Miner_SurpriseExpiration,
-			Sectors:   nil,
+		if rt.CurrEpoch() <= st.PoStState.ProvingPeriodStart {
+			rt.Abort(exitcode.ErrIllegalState, "Not currently in submission window for PoSt")
 		}
-		surpriseDuration := power.SurprisePostChallengeDuration
-		a.enrollCronEvent(rt, rt.CurrEpoch()+surpriseDuration, &cronPayload)
-	}
-	return &adt.EmptyValue{}
-}
+		a.verifyWindowedPost(rt, &st, params)
 
-type SubmitSurprisePoStResponseParams struct {
-	onChainInfo abi.OnChainSurprisePoStVerifyInfo
-}
-
-// Invoked by miner's worker address to submit a response to a pending SurprisePoSt challenge.
-func (a Actor) SubmitSurprisePoStResponse(rt Runtime, params *SubmitSurprisePoStResponseParams) *adt.EmptyValue {
-	var st State
-	rt.State().Transaction(&st, func() interface{} {
-		rt.ValidateImmediateCallerIs(st.Info.Worker)
-		if !st.PoStState.isChallenged() {
-			rt.Abort(exitcode.ErrIllegalState, "Not currently challenged")
-		}
-		a.verifySurprisePost(rt, &st, &params.onChainInfo)
+		// TODO: post is valid, do we do anything with faults now?
+		// Assume there will be work to be done here once faults are
+		// implemented, with faults, we have potential power adjustments
 
 		st.PoStState = PoStState{
-			LastSuccessfulPoSt:     rt.CurrEpoch(),
-			SurpriseChallengeEpoch: epochUndefined,
+			ProvingPeriodStart:     st.PoStState.ProvingPeriodStart + ProvingPeriod,
 			NumConsecutiveFailures: 0,
 		}
+
+		st.ProvingSet = st.Sectors
```

Comment:
This isn't wrong, but it should *also* happen when the challenge window opens. Deferring it until then so we only do it once would make that clear.

Reply:
I agree with this @anorth (and it would be nicer to only do it on challenge window open rather than every time the sector set changes), but the challenge window opening is no longer an actual event, so unless we add that back (ie -- have it triggered by power actor -- i wrote something to that effect below), this is the way to do it I think.

Reply:
@anorth it doesnt need to happen again at that point, once we submit the post, it bumps our window forward in time, so that we're no longer inside it

Reply:
Ok, I understand now that because there is no cron callback at the start of the window we can't snapshot then, and must maintain proving set throughout the dead period between challenge windows. In that case, we must be careful to keep them in sync (I see you've correctly done the same thing at terminateSector).

Reply:
ProvingSet will be updated again (aka by removing the faulty sectors) when a Challenge window begins right?

Reply:
@zixuanzh no I am not expecting that change. We're taking the cheap copy, will subtract faults on the fly when evaluating. 

But, I think the fault set needs to be frozen when the window opens. so maybe we'll need that cron callback after all. probably for a follow-up

Reply:
see #143